test/sqllogictest/scalar-func-table-position.slt

```diff
@@ -0,0 +1,158 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Selecting from scalar functions in FROM clauses
```

Comment:
Can you add something like `SELECT length FROM length('str' || (select length FROM length('str')));` here too?

Also interested in seeing these joined with a more typical source:

```
SELECT * FROM length('str'), (values (1, 2), (4, 5)) v;
```

Reply:
Great calls, ty––added those, and they worked as expected w/o further code changes.